airavata-services/profile-service/iam-admin-services-core/src/main/java/org/apache/airavata/service/profile/iam/admin/services/core/impl/TenantManagementKeycloakImpl.java

```diff
@@ -0,0 +1,254 @@
+                credential.setTemporary(true);
+                retrievedUser.resetPassword(credential);
+                List<ClientRepresentation> realmClients = client.realm(gatewayDetails.getGatewayId()).clients().findAll();
+                String realmManagementClientId=null;
+                for(ClientRepresentation realmClient : realmClients){
+                    if(realmClient.getClientId().equals("realm-management")){
+                        realmManagementClientId = realmClient.getId();
+                    }
+                }
+                retrievedUser.roles().clientLevel(realmManagementClientId).add(retrievedUser.roles().clientLevel(realmManagementClientId).listAvailable());
+                return true;
+            } else {
+                logger.error("Request for Tenant Admin Account Creation failed with HTTP code : " + httpResponse.getStatus());
+                logger.error("Reason for Tenant Admin account creation failure : " + httpResponse.getStatusInfo());
+                return false;
+            }
+        }catch (ApplicationSettingsException ex) {
+            logger.error("Error getting values from property file, reason: " + ex.getCause(), ex);
+            IamAdminServicesException exception = new IamAdminServicesException();
+            exception.setMessage("Error getting values from property file, reason " + ex.getMessage());
+            throw exception;
+        }catch (Exception ex){
+            logger.error("Error creating Realm Admin Account in keycloak server, reason: " + ex.getCause(), ex);
+            IamAdminServicesException exception = new IamAdminServicesException();
+            exception.setMessage("Error creating Realm Admin Account in keycloak server, reason: " + ex.getMessage());
+            throw exception;
+        }
+    }
+
+    @Override
+    public Gateway configureClient(PasswordCredential isSuperAdminPasswordCreds, Gateway gatewayDetails) throws IamAdminServicesException{
+        try{
+            Keycloak client = TenantManagementKeycloakImpl.getClient(ServerSettings.getIamServerUrl(), "master", isSuperAdminPasswordCreds);
+            ClientRepresentation pgaClient = new ClientRepresentation();
+            pgaClient.setName("pga");
+            pgaClient.setClientId("pga");
+            pgaClient.setProtocol("openid-connect");
+            pgaClient.setStandardFlowEnabled(true);
+            pgaClient.setEnabled(true);
+            pgaClient.setAuthorizationServicesEnabled(true);
+            pgaClient.setDirectAccessGrantsEnabled(true);
+            pgaClient.setServiceAccountsEnabled(true);
+            pgaClient.setFullScopeAllowed(true);
+            pgaClient.setClientAuthenticatorType("client-secret");
+            String[] defaultRoles = {"gateway-user"};
+            pgaClient.setDefaultRoles(defaultRoles);
+            List<String> redirectUris = new ArrayList<>();
+            redirectUris.add("http://accord.scigap.org/callback-url");
```

Comment:
@marcus,
The Gateway profile already has this GatewayUrl field,
[Gateway Profile](https://github.com/apache/airavata/blob/master/thrift-interface-descriptions/data-models/experiment-catalog-models/workspace_model.thrift)
```
struct Gateway {
    1: required string gatewayId,
    2: required GatewayApprovalStatus gatewayApprovalStatus,
    3: optional string gatewayName,
    4: optional string domain,
    5: optional string emailAddress
    6: optional string gatewayAcronym,
    7: optional string gatewayURL,
    8: optional string gatewayPublicAbstract,
    9: optional string reviewProposalDescription,
    10: optional string gatewayAdminFirstName,
    11: optional string gatewayAdminLastName,
    12: optional string gatewayAdminEmail,
    13: optional string identityServerUserName,
    14: optional string identityServerPasswordToken,
    15: optional string declinedReason,
    16: optional string oauthClientId,
    17: optional string oauthClientSecret,
    18: optional i64 requestCreationTime,
    19: optional string requesterUsername
}
```

Can i use it to construct callback url. i.e, append callback-url to that field ?